brainframe/cli/main.py

```diff
@@ -0,0 +1,73 @@
+from argparse import ArgumentParser
+import sys
+from pathlib import Path
+import os
+import signal
+
+import i18n
+
+from brainframe.cli import (
+    print_utils,
+    docker_compose,
+    commands,
+    defaults,
+    env_vars,
+)
+
+
+def main():
+    i18n.load_path.append(_TRANSLATIONS_PATH)
+
+    parser = ArgumentParser(
+        description=i18n.t("portal.description"), usage=i18n.t("portal.usage")
+    )
+
+    parser.add_argument(
+        "command", default=None, nargs="?", help=i18n.t("portal.command-help")
+    )
+
+    # Set environment variables to their default values if unset
+    if env_vars.INSTALL_PATH not in os.environ:
+        os.environ[env_vars.INSTALL_PATH] = str(defaults.INSTALL_PATH)
+    # This environment variable must be set as it is used by the
+    # docker-compose.yml to find the data path to volume mount
+    if env_vars.DATA_PATH not in os.environ:
+        os.environ[env_vars.DATA_PATH] = str(defaults.DATA_PATH)
+
+    install_path = Path(os.environ[env_vars.INSTALL_PATH])
+    data_path = Path(os.environ[env_vars.DATA_PATH])
+
+    args = parser.parse_args(sys.argv[1:2])
+
+    # Exit with a clean error when interrupted
+    def on_sigint(_sig, _frame):
+        print("")
+        print_utils.fail_translate("portal.interrupted")
+
+    signal.signal(signal.SIGINT, on_sigint)
+
+    if args.command is None:
+        print_utils.translate("portal.no-command-provided")
+    elif args.command == "install":
+        commands.install()
+    elif args.command == "backup":
```

Comment:
Suggestion- a decorator for making these commands, something like
@register_command("command_name") or something that would populate a dict that `main` would use.

Reply:
Ooh, fancy! I'm down.